infra/feast-operator/internal/controller/services/services.go

```diff
@@ -499,6 +499,13 @@ func (feast *FeastServices) getContainerCommand(feastType FeastServiceType) []st
 	deploySettings := FeastServiceConstants[feastType]
 	targetPort := deploySettings.TargetHttpPort
 	tls := feast.getTlsConfigs(feastType)
+
+	if feastType == RegistryFeastType {
+		registry := feast.Handler.FeatureStore.Spec.Services.Registry
+		if registry != nil && registry.Local != nil && registry.Local.Server != nil && registry.Local.Server.RestAPIEnabled {
```

Comment:
Use the built-in method for checking if Local registry is set & the applied status when verifying whether to act on a field. This is because applied status will also have defaults set, etc.
```suggestion
	if feastType == RegistryFeastType && feast.isLocalRegistry() {
		registry := feast.Handler.FeatureStore.Status.Applied.Services.Registry
		if registry.Local.Server.RestAPIEnabled {
```